semtypes/src/main/java/io/ballerina/semtype/definition/Field.java

```diff
@@ -34,4 +39,20 @@
     public static Field from(String name, SemType type) {
         return new Field(name, type);
     }
+
+    public static SplitFieldHolder splitFields(List<Field> fields) {
```

Comment:
Shall we move this and `fieldName` to MappingDefinition. This looks like a helper function for that.